FluentTerminal.SystemTray/Services/TerminalsManager.cs

```diff
@@ -8,36 +9,97 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Text;
+using System.Text.RegularExpressions;
 using Windows.ApplicationModel;
 
 namespace FluentTerminal.SystemTray.Services
 {
+    public struct TerminalSessionInfo
+    {
+        public DateTime StartTime { get; set; }
+        public string ProfileName { get; set; }
+        public ITerminalSession Session { get; set; }
+    }
+
     public class TerminalsManager
     {
-        private readonly Dictionary<byte, ITerminalSession> _terminals = new Dictionary<byte, ITerminalSession>();
+        private readonly Dictionary<byte, TerminalSessionInfo> _terminals = new Dictionary<byte, TerminalSessionInfo>();
 
         public event EventHandler<TerminalOutput> DisplayOutputRequested;
 
         public event EventHandler<TerminalExitStatus> TerminalExited;
 
+        private readonly ISettingsService _settingsService;
+
+        private static readonly Regex EscapeSequencePattern = new Regex(@"((\x9B|\x1B\[)[0-?]*[ -\/]*[@-~])|((\x9D|\x1B\]).*\x07)", RegexOptions.Compiled);
+
+        private Dictionary<byte, string> _cachedLogPath = new Dictionary<byte, string>();
+
+        public TerminalsManager(ISettingsService settingsService)
+        {
+            _settingsService = settingsService;
+        }
+
         public void DisplayTerminalOutput(byte terminalId, byte[] output)
         {
+            if (_settingsService.GetApplicationSettings().EnableLogging && Directory.Exists(_settingsService.GetApplicationSettings().LogDirectoryPath))

```

Comment:
Maybe we should cache this. Might cause up to two disk reads on every output

Reply:
@felixse good catch! have rebased whole changelist, thanks